test/lib/ufe/testUngroupCmd.py

```diff
@@ -24,17 +24,55 @@
 
 import mayaUsd.ufe
 
-from pxr import Kind
-from pxr import Usd
+from pxr import UsdGeom, Gf
 
+from maya.api import OpenMaya as OpenMaya
 from maya import cmds
 from maya import standalone
 
 import ufe
-
 import os
 import unittest
 
+def createStage():
+    ''' create a simple stage '''
+    cmds.file(new=True, force=True)
+    import mayaUsd_createStageWithNewLayer
+    proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+    proxyShapes = cmds.ls(type="mayaUsdProxyShapeBase", long=True)
+    proxyShapePath = proxyShapes[0]
+    proxyShapeItem = ufe.Hierarchy.createItem(ufe.PathString.path(proxyShapePath))
+    proxyShapeContextOps = ufe.ContextOps.contextOps(proxyShapeItem)
+    stage = mayaUsd.lib.GetPrim(proxyShapePath).GetStage()
+    return (stage, proxyShapePathStr, proxyShapeItem, proxyShapeContextOps)
+
+class SphereGenerator():
+    ''' simple sphere generator '''
+    def __init__(self, num, contextOp, proxyShapePathStr):
+        self.gen = self.__generate(num)
+        self.num = num
+        self.contextOp = contextOp
+        self.proxyShapePathStr = proxyShapePathStr
+
+    def createSphere(self):
+        return next(self.gen)
+
+    def __addPrimSphere(self, incrment):
+        self.contextOp.doOp(['Add New Prim', 'Sphere'])
```

Comment:
You are right, UsdStage::DefinePrim() would do the same but at the time I wasn't thinking about it. I would like to leave it as since it does the work.